lib/src/webcrypto/webcrypto.ecdh.dart

```diff
@@ -14,43 +14,231 @@
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// Future<void> main() async {
+  ///   // Generate a new key pair using the P-256 curve.
+  ///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+  /// 
+  ///   // Export the private key.
+  ///   final exportedPrivateKey = await keyPair.privateKey.exportJsonWebKey();
+  ///   print(exportedPrivateKey);
+  /// 
+  ///   // Export the public key.
+  ///   final exportedPublicKey = await keyPair.publicKey.exportJsonWebKey();
+  ///   print(exportedPublicKey);
+  /// }
+  /// ```
+  /// 
   static Future<KeyPair<EcdhPrivateKey, EcdhPublicKey>> generateKey(
     EllipticCurve curve,
   ) {
     return impl.ecdhPrivateKey_generateKey(curve);
   }
 
+  /// Derive an array of bits from the [EcdhPrivateKey].
+  /// The [length] parameter specifies the number of bits to derive and 
+  /// should be multiples of 8.
+  /// The [publicKey] parameter is the [EcdhPublicKey].
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// Future<void> main() async {
+  ///   // Generate a new key pair using the P-256 curve.
+  ///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+  /// 
+  ///   // Derive 256 bits from the private key using the public key.
+  ///   final derivedBits = await keyPair.privateKey.deriveBits(256, keyPair.publicKey);
+  ///   print(derivedBits);
+  /// }
+  /// ```
+  Future<Uint8List> deriveBits(int length, EcdhPublicKey publicKey);
   // Note some webcrypto implementations (chrome, not firefox) supports passing
   // null for length (in this primitive). However, you can always know the right
   // length from the curve. Note p512 can provide up to: 528 bits!!!
   //
   // See: https://tools.ietf.org/html/rfc6090#section-4
   // Notice that this is not uniformly distributed, see also:
   // https://tools.ietf.org/html/rfc6090#appendix-B
-  Future<Uint8List> deriveBits(int length, EcdhPublicKey publicKey);
 
-  // Note. unsupported on Firefox, see EcdsaPrivateKey.importPkcs8Key
+  /// Export the [EcdhPrivateKey] as a PKCS8 key.
+  /// This returns the private key as an octet string.
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// Future<void> main() async {
+  ///   // Generate a new key pair using the P-256 curve.
+  ///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+  /// 
+  ///   // Export the private key (print it in same format as it was given).
+  ///   final exportedPkcs8Key = await privateKey.exportPkcs8Key();
+  ///   print(exportedPkcs8Key);
+  /// }
+  /// ```
+  /// 
+  // Note. unsupported on Firefox
   Future<Uint8List> exportPkcs8Key();
 
+  /// Export the [EcdhPrivateKey] and [EcPublicKey] as a [JSON Web Key][1].
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
```

Comment:
Check documentation for the other `exportJsonWebKey` methods. There is probably a macro describing what comes out.

Reply:
Added ```{@macro exportJsonWebKey:returns}```

Reply:
Perfect!